### PR TITLE
Request a BOXUser object by user ID

### PR DESCRIPTION
**Pull request by jdizoglio**

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @jdizoglio has signed the CLA. Thanks for the pull request!

File: BoxContentSDK/BoxContentSDK/BOXContentSDKConstants.m [hunk 1]
```diff
@@ -107,6 +107,8 @@
 BOXSharedLinkAccessLevel *const BOXSharedLinkAccessLevelOpen = @"open";
 BOXSharedLinkAccessLevel *const BOXSharedLinkAccessLevelCompany = @"company";
 BOXSharedLinkAccessLevel *const BOXSharedLinkAccessLevelCollaborators = @"collaborators";
+BOXSharedLinkAccessLevel *const BOXSharedLinkExpireLinkFileAccess = @"expireLinkFiles";
```
Comment (thuyBox) [inline BoxContentSDK/BoxContentSDK/BOXContentSDKConstants.m]: don't seem like we use them?

File: BoxContentSDK/BoxContentSDK/BOXContentSDKConstants.h [hunk 2]
```diff
@@ -117,6 +117,8 @@ typedef NSString BOXSharedLinkAccessLevel;
 extern BOXSharedLinkAccessLevel *const BOXSharedLinkAccessLevelOpen;
 extern BOXSharedLinkAccessLevel *const BOXSharedLinkAccessLevelCompany;
 extern BOXSharedLinkAccessLevel *const BOXSharedLinkAccessLevelCollaborators;
+extern BOXSharedLinkAccessLevel *const BOXSharedLinkExpireLinkFileAvailable;
```
Comment (boxtcarpel) [inline BoxContentSDK/BoxContentSDK/BOXContentSDKConstants.h]: These aren't quite shared link access levels...
Should these just be separate constants specific to the expiration date?

Comment (boxtcarpel) [reply BoxContentSDK/BoxContentSDK/BOXContentSDKConstants.h]: In fact, these actually aren't used anywhere either, can we remove them?

File: BoxContentSDK/BoxContentSDK/BOXContentSDKConstants.m (hunk 1 above)
Comment (boxtcarpel) [inline BoxContentSDK/BoxContentSDK/BOXContentSDKConstants.m]: This constant name doesn't actually match the definition above, and is not used anywhere. Can we just remove these?

File: BoxContentSDK/BoxContentSDK/Requests/BOXUserRequest.h [hunk 3]
```diff
@@ -14,6 +14,10 @@
  */
 @property (nonatomic, readwrite, assign) BOOL requestAllUserFields;
 
+@property (nonatomic, readonly, strong) NSString *userID;
```
Comment (boxtcarpel) [inline BoxContentSDK/BoxContentSDK/Requests/BOXUserRequest.h:17]: Should we leave a comment that this property only reports a value if a specific user is requested, and that in the default case where no user is specified the current user is used (but this value will be nil or an empty string)?

**Review by boxtcarpel — APPROVED** [review]

File: BoxContentSDK/BoxContentSDK/BOXContentSDKConstants.m (hunk 1 above)
Comment (jdizoglio) [reply BoxContentSDK/BoxContentSDK/BOXContentSDKConstants.m]: removed

File: BoxContentSDK/BoxContentSDK/BOXContentSDKConstants.h (hunk 2 above)
Comment (jdizoglio) [reply BoxContentSDK/BoxContentSDK/BOXContentSDKConstants.h]: removed

File: BoxContentSDK/BoxContentSDK/BOXContentSDKConstants.m (hunk 1 above)
Comment (jdizoglio) [reply BoxContentSDK/BoxContentSDK/BOXContentSDKConstants.m]: removed

File: BoxContentSDK/BoxContentSDK/Requests/BOXUserRequest.h (hunk 3 above)
Comment (jdizoglio) [reply BoxContentSDK/BoxContentSDK/Requests/BOXUserRequest.h:17]: Good point, for a manual construction of the request, setting the userID will have the same impact as the initWithUserID

**Pull request merged**